tools/run_tests/performance/loadtest_config.py

```diff
@@ -118,17 +118,72 @@ def gen_run_indices(runs_per_test: int) -> Iterable[str]:
         yield index_fmt.format(i)
 
 
+def scenario_name(base_name, client_channels, server_threads, offered_load):
+    elements = [base_name]
+    if client_channels:
+        elements.append(f'{client_channels}channels')
+    if server_threads:
+        elements.append(f'{server_threads}threads')
+    if offered_load:
+        elements.append(f'{offered_load}load')
+    return '_'.join(elements)
+
+
+def scenario_transform_function(
+    client_channels: int, server_threads: int, offered_loads: Iterable[int]
+) -> Optional[Callable[[Iterable[Mapping[str, Any]]], Iterable[Mapping[str,
+                                                                       Any]]]]:
+    """Returns a transform to be applied to a list of scenarios."""
+    if not any([client_channels, server_threads, len(offered_loads)]):
+        return lambda s: s
+
+    def _transform(
+            scenarios: Iterable[Mapping[str,
+                                        Any]]) -> Iterable[Mapping[str, Any]]:
+        """Transforms scenarios by inserting num of client channels, number of async_server_threads and offered_load."""
+
+        for base_scenario in scenarios:
+            base_name = base_scenario['name']
+            if client_channels:
+                base_scenario['client_config'][
+                    'client_channels'] = client_channels
+            if server_threads:
+                base_scenario['server_config'][
+                    'async_server_threads'] = server_threads
+            if not offered_loads:
```

Comment:
Nit: I would add a blank line before this line and after "return", to make it clear that it breaks there. Probably do the same for the two if block above it as well.